metapool/prep.py

```diff
@@ -461,6 +481,103 @@ def preparations_for_run(run_path, sheet, pipeline='fastp-and-minimap2'):
+
+    if not_present:
+        raise ValueError("Required columns are missing: %s" %
+                         ', '.join(not_present))
+
+    for project, project_sheet in mapping_file.groupby('Project_name'):
+        project_name = remove_qiita_id(project)
+        qiita_id = project.replace(project_name + '_', '')
+
+        # if the Qiita ID is not found then make for an easy find/replace
+        if qiita_id == project:
+            qiita_id = 'QIITA-ID'
+
+        for lane, lane_sheet in project_sheet.groupby('lane'):
+            lane_sheet = lane_sheet.set_index('sample_name')
+
+            # this is the portion of the loop that creates the prep
+            data = []
+
+            for sample_id, sample in lane_sheet.iterrows():
+                run_prefix = get_run_prefix(run_path,
+                                            project,
+                                            sample_id,
+                                            lane,
+                                            'fastp-and-minimap2')
+
+                # we don't care about the sample if there's no file
+                if run_prefix is None:
+                    continue
+
+                row = {c: '' for c in PREP_MF_COLUMNS}
+
+                row["sample_name"] = sample.Sample_ID
+                row["experiment_design_description"] = \
+                    sample.EXPERIMENT_DESIGN_DESCRIPTION
+                row["library_construction_protocol"] = \
+                    sample.LIBRARY_CONSTRUCTION_PROTOCOL
+                row["platform"] = sample.PLATFORM
+                row["run_center"] = sample.RUN_CENTER
+                row["run_date"] = sample.RUN_DATE
+                # run_prefix will be set to the value determined above,
+                # rather than what was in the mapping-file.
+                row["run_prefix"] = run_prefix
+                row["sequencing_meth"] = sample.sequencing_meth
+                row["center_name"] = sample.center_name
+                row["center_project_name"] = sample.center_project_name
+                row["instrument_model"] = sample.INSTRUMENT_MODEL
+                row["runid"] = run_id
+                row["sample_plate"] = sample.Sample_Plate
+                row["lane"] = lane
+                row["sample_project"] = project_name
+                data.append(row)
+
+            if not data:
+                warnings.warn('Project %s and Lane %s have no data' %
+                              (project, lane), UserWarning)
+
+            # the American Gut Project is a special case. We'll likely continue
+            # to grow this study with more and more runs. So we fill some of
+            # the blanks if we can verify the study id corresponds to the AGP.
+            # This was a request by Daniel McDonald and Gail
+
+            prep = agp_transform(pd.DataFrame(columns=PREP_COLUMNS,
+                                              data=data),
+                                 qiita_id)
```

Comment:
Just to confirm, is this also done in the metagenomic block?

Reply:
Yup:
https://github.com/charles-cowart/metagenomics_pooling_notebook/blob/1b88781aa8030402ce812a0791d783ceaaff66ed/metapool/prep.py#L468-L471